Final Project Submission - Christina

christina-baldwin opened the pull request.

christina-baldwin committed:
setting up my pages

christina-baldwin committed:
updating the readme

christina-baldwin committed:
setting up routes

christina-baldwin committed:
removing un-needed files and adding the notfound page

christina-baldwin committed:
setting up folders for authorisation and setting up the user model

christina-baldwin committed:
setting up the auth.js route

christina-baldwin committed:
setting up the auth.js middleware

christina-baldwin committed:
connecting to my mongo database

christina-baldwin committed:
fixing some file changes and npm installs

christina-baldwin committed:
fixing dotenc config

christina-baldwin committed:
setting up login and connecting it

christina-baldwin committed:
setting up registration page

christina-baldwin committed:
typo fix

christina-baldwin committed:
todo message added

christina-baldwin committed:
setting up tailwind

christina-baldwin committed:
removing unnecessary imports

christina-baldwin committed:
basic styling

christina-baldwin committed:
line height

christina-baldwin committed:
adding a logo

christina-baldwin committed:
basic styling for login

christina-baldwin committed:
basic styling for registration

christina-baldwin committed:
todo notes

christina-baldwin committed:
adding register and login validation on the frontned

christina-baldwin committed:
validation messages styling

christina-baldwin committed:
secret key issues and authorisation

christina-baldwin committed:
so my dahsboard shows

christina-baldwin committed:
adding the nav

christina-baldwin committed:
adding a userinfo route

christina-baldwin committed:
fixing some import errors

christina-baldwin committed:
getting the name of the user to show in the nav

christina-baldwin committed:
getting the logout to work

christina-baldwin committed:
creating and adding a sidebar

christina-baldwin committed:
organising the structure of my dashboard

christina-baldwin committed:
setting up photo upload routes on the backend

christina-baldwin committed:
file upload on the frontend from and existing file

christina-baldwin committed:
adding the sidebar to the photo upload page

christina-baldwin committed:
adding the sidebar to the other pages

christina-baldwin committed:
adding a desk schema

christina-baldwin committed:
saving a new desk to the database

christina-baldwin committed:
showing latest upload on the uploads page

christina-baldwin committed:
adding delete photo functionality on backend and frontend

christina-baldwin committed:
editing notes and removing unused imports

christina-baldwin committed:
showing the latest upload preview on my dashboard landing page

christina-baldwin committed:
renaming

christina-baldwin committed:
adding a conditional for when there are suggestions already generated

christina-baldwin committed:
setting up a button to take a photo

christina-baldwin committed:
changing back and fixing wording

christina-baldwin committed:
setting up the settings page structure

christina-baldwin committed:
starting the editing functionality. TODO patch on the backend

christina-baldwin committed:
patch on the backend and getting it to work with change name

christina-baldwin committed:
styling changes

christina-baldwin committed:
adding the change email functionality

christina-baldwin committed:
password change

christina-baldwin committed:
making password change clearer

christina-baldwin committed:
styling the settings

christina-baldwin committed:
setting up the ai gnerated suggestions backend route

christina-baldwin committed:
setting up so the ai generates suggestions when clicking the button on the frontend

christina-baldwin committed:
having the desk photo and desk suggestions show on the frontend

christina-baldwin committed:
adding a message to be disoplayed and a nnote to maybe change it later

christina-baldwin committed:
updating my ai text prompt

christina-baldwin committed:
changing the way the ai prompt returns text

christina-baldwin committed:
adding an input to put in your personal desk problems for the ai to analyse and give better suggestions

christina-baldwin committed:
making sidabar titles clearer

christina-baldwin committed:
also displaying old suggestions

christina-baldwin committed:
spacing of suggestions

christina-baldwin committed:
adding notes and placeholders for new features

christina-baldwin committed:
styling

christina-baldwin committed:
styling & addinbg placeholder to add/edit problems after upload

christina-baldwin committed:
getting popup functionility to work

christina-baldwin committed:
hide/load older suggestions

christina-baldwin committed:
spacing under image

christina-baldwin committed:
adding a patch request and frontend functionality for the add/edit desk problems

christina-baldwin committed:
button wording that makes more sense

christina-baldwin committed:
starting the main styling

christina-baldwin committed:
figuring out styling for the login page

christina-baldwin committed:
styling registration to lok like login

christina-baldwin committed:
messing around with base colours and styling

christina-baldwin committed:
finialising design for landing page

christina-baldwin committed:
setting up colour classes in tailwind for base theme colours

christina-baldwin committed:
removing old config

christina-baldwin committed:
adding the font families in the theme

christina-baldwin committed:
styling the login page

christina-baldwin committed:
styling registration page

christina-baldwin committed:
styling the nav

christina-baldwin committed:
sidebar styling

christina-baldwin committed:
styling the dashboard

christina-baldwin committed:
styling the upload page

christina-baldwin committed:
styling the edit problems view

christina-baldwin committed:
fixed width and height for dashboard, styling the suggestions to match

christina-baldwin committed:
styling the settings with the same theme

christina-baldwin committed:
setting up placeholders for dashboard home

christina-baldwin committed:
updating the style of the user message so its more visible

christina-baldwin committed:
delete confirmation popu

christina-baldwin committed:
showing the last logged in date/time stamp

christina-baldwin committed:
formatting so the last logged in reads well for users

christina-baldwin committed:
displaying the days since desk photo was uploaded

christina-baldwin committed:
conditional chaining for when desk photo was uploaded to stop an error from throwing

christina-baldwin committed:
creating and displaying my quick summary for my exisitng hobby desk

christina-baldwin committed:
styling the summary

christina-baldwin committed:
summary title

christina-baldwin committed:
implementing drag and drop for photo upload

christina-baldwin committed:
wording

christina-baldwin committed:
editying the landing page blurb

christina-baldwin committed:
making login and register look cohesive

christina-baldwin committed:
changing the focus ring colours

christina-baldwin committed:
wording

christina-baldwin committed:
making the ai prompt more specific

christina-baldwin committed:
changing some phrasing

christina-baldwin committed:
cahanging naming

christina-baldwin committed:
adding a generating suggestions message

christina-baldwin committed:
adding validation to only allow image files with drag-drop

christina-baldwin committed:
rewriting the tips popup so its a bit more instructional

christina-baldwin committed:
moving the help popup prompt further up

christina-baldwin committed:
adding a lightbulb icon to my tips popup link

christina-baldwin committed:
making titles bigger

christina-baldwin committed:
replacing with the onrender url

christina-baldwin committed:
added dist to gitignore

christina-baldwin committed:
displaying user messages better

christina-baldwin committed:
protecting my non-public routes

christina-baldwin committed:
fixing list elements to be more acessible

christina-baldwin committed:
changing the focus colours to match the theme

christina-baldwin committed:
resposnive design for landing page

christina-baldwin committed:
responsive design for login

christina-baldwin committed:
responsive design for register page

christina-baldwin committed:
responsive styling the sidebar

christina-baldwin committed:
responsive design for dashboard home page

christina-baldwin committed:
responsive design for the upload page

christina-baldwin committed:
responsive design for the suggestions page

christina-baldwin committed:
responsive design for settings page

christina-baldwin committed:
adding a footer

christina-baldwin committed:
fixing some styling

christina-baldwin committed:
isloading state for register and login

christina-baldwin committed:
fixing the saving error when editing problems

christina-baldwin committed:
more specific validation

christina-baldwin committed:
styling and hero img change

christina-baldwin committed:
making sure problems is saved on backedn at upload

christina-baldwin committed:
some changes (temporary, will split this up)

christina-baldwin committed:
setting up for refactoring the upload page into components

christina-baldwin committed:
setting up zustand for my latest desk

christina-baldwin committed:
fixing the zustand import

christina-baldwin committed:
creating and incorporating the UploaDesk component

christina-baldwin committed:
finalising and incoporating the LatestDesk component into Upload page

christina-baldwin committed:
removing un-needed code

christina-baldwin committed:
removing comments

christina-baldwin committed:
using the deskstore on the dashboard home

christina-baldwin committed:
changin the wording

christina-baldwin committed:
using the motion library to animate

christina-baldwin committed:
removing unwanted notes

christina-baldwin committed:
fixing some responsive design widths

christina-baldwin committed:
internal scroll in componente

christina-baldwin committed:
closing popup on escape and when clicking out of the popup, fixing popup responsive design

christina-baldwin committed:
showing all older suggestions and fixing bug when suggestions for latest havent been generated yet

christina-baldwin committed:
adding dates for when desks were uploaded

christina-baldwin committed:
adding a filter for older desks

christina-baldwin committed:
link that goes back to upload from suggestions if user wants to generate/regenerate suggestions

christina-baldwin committed:
validation on each form input

christina-baldwin committed:
adding an isUploading state

christina-baldwin committed:
adding notes to help with my demo

christina-baldwin committed:
adding a message for when the desk is deleted successfully

christina-baldwin committed:
adding text if suggestions arent available for older desks

christina-baldwin committed:
fixing some styling and removing notes

christina-baldwin committed:
typo fix

christina-baldwin committed:
updating so image shows on netlify

christina-baldwin committed:
removing svg

christina-baldwin committed:
fixing padding

christina-baldwin committed:
smaller gap in settings

christina-baldwin committed:
Update README.md

christina-baldwin committed:
adding a favicon

christina-baldwin committed:
Merge branch 'main' of https://github.com/christina-baldwin/desk-forge

christina-baldwin committed:
adding an ai call limit

christina-baldwin committed:
making the ai call limit 10

christina-baldwin committed:
changed the ai call limit message

christina-baldwin committed:
adding a max number of accounts that are allowed

JennieDalgren reviewed (changes requested):
UX fix?
A little bit hard to see that I clicked the register button
<img width="633" height="490" alt="image" src="https://github.com/user-attachments/assets/622ae489-6914-4e6e-abee-86c6c2d9a38d" />
I had to scroll down. You could change the button text just as you do on uploading... for example.

Also, I can't generate a suggestion
<img width="1512" height="982" alt="image" src="https://github.com/user-attachments/assets/ebd4ec3e-f166-4972-981a-b0f86bf38899" />

Please take a look and let me know when it is working so I can continue the grading.

christina-baldwin commented:
Thanks for the suggestion! I will work on the UX suggestion this weekend probably as I have a full week! Hope that's ok. 

About the generation of suggestions, it works fine for me so im not really sure what the issue is, maybe it was an onrender thing? If you try again now does it work?

JennieDalgren reviewed (commented):
Still the same, some kind of server issue

<img width="1512" height="982" alt="image" src="https://github.com/user-attachments/assets/cd3abb03-f2b3-4962-a345-7593422fd2b9" />

christina-baldwin commented:
Ah thats weird. Seems the image preview isn't showing either. I wont have time to look at it before Friday afternoon as i have my LIA and volunteering these few days. Hope that isn't an issue? But just in case, did you try and manage to generate on this account before?

JennieDalgren commented:
tried with different account, never managed to generate suggestions.
No worries with finishing now this week! just ping me when you have had time to look at it.

christina-baldwin committed:
logs and not allowing certain file formats

christina-baldwin committed:
fix

christina-baldwin commented:
ok so, i couldn't replicate the error you're getting at all, sent it to a friend, tried different browsers etc, live, local, etc. but i had a thought that maybe you were uploading an unsupported file format, so i added a check and stop for that just in case, i also added a bunch of logs to see if that tells me anything, can you also please walk me through your exact steps that got you there so what image you're using etc.

christina-baldwin committed:
redirects to login after successful registration

JennieDalgren commented:
This is the error I get when trying to generate suggestions:
<img width="1502" height="919" alt="image" src="https://github.com/user-attachments/assets/fc14039b-f9e5-4f5f-967b-416fc4f067e0" />

account: **poppy@p.com**
password: **poppy123**

christina-baldwin commented:
Ok so I logged into your account (thanks for providing those details), and that image fails to generate for me too, but, I uploaded a different image and that worked. I'm pretty sure it has something to do with the image but I'm not sure what, if you have any ideas I'm open. But it could be size and/or resolution thing or a file tyoe somehow as I see it downloads as a .avif file but i dont know how that gets through the check. I will try limit that further, if you have details on the image or you are using, or ideas, that could help

christina-baldwin committed:
adding stricter checks for file sizes and types

HIPPIEKICK reviewed (approved):
I'll approve this now since you have put in the work + it never fails for me. Good job! ⭐

christina-baldwin commented:
Aw thank you both! ✨️ I'd love to figure out this bug though so if its still doing it for Jennie and you guys have time/ideas please let me know!
